modules/gdscript/doc_classes/@GDScript.xml

```diff
@@ -288,6 +288,17 @@
 				[/codeblock]
 			</description>
 		</method>
+		<method name="deep_equal">
+			<return type="bool">
+			</return>
+			<argument index="0" name="a" type="Variant">
+			</argument>
+			<argument index="1" name="b" type="Variant">
+			</argument>
+			<description>
+				Compares two values, but, in contrast to `==`, any `Dictionary` or `Array` found is compared recursively, checking its actual contents.
```

Comment:
we may want to stress that `==` only does pointer comparison here, otherwise the user could think `==` does a kind of non recursive comparison (e.g. only comparing values of an array if those values are not arrays themselves)

Reply:
Good point. I've written a much more thorough explanation.